tests/components/accuweather/test_sensor.py

```diff
@@ -0,0 +1,598 @@
+    assert state.attributes.get(ATTR_ICON) == "mdi:weather-fog"
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == LENGTH_METERS
+
+    entry = registry.async_get("sensor.home_cloud_ceiling")
+    assert entry
+    assert entry.unique_id == "0123456-ceiling"
+
+    state = hass.states.get("sensor.home_precipitation")
+    assert state
+    assert state.state == "0.0"
+    assert state.attributes.get(ATTR_ATTRIBUTION) == ATTRIBUTION
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == LENGTH_MILIMETERS
+    assert state.attributes.get(ATTR_ICON) == "mdi:weather-rainy"
+    assert state.attributes.get("type") is None
+
+    entry = registry.async_get("sensor.home_precipitation")
+    assert entry
+    assert entry.unique_id == "0123456-precipitation"
+
+    state = hass.states.get("sensor.home_pressure_tendency")
+    assert state
+    assert state.state == "falling"
+    assert state.attributes.get(ATTR_ATTRIBUTION) == ATTRIBUTION
+    assert state.attributes.get(ATTR_ICON) == "mdi:gauge"
+    assert state.attributes.get(ATTR_DEVICE_CLASS) == "accuweather__pressure_tendency"
+
+    entry = registry.async_get("sensor.home_pressure_tendency")
+    assert entry
+    assert entry.unique_id == "0123456-pressuretendency"
+
+    state = hass.states.get("sensor.home_realfeel_temperature")
+    assert state
+    assert state.state == "25.1"
+    assert state.attributes.get(ATTR_ATTRIBUTION) == ATTRIBUTION
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == TEMP_CELSIUS
+    assert state.attributes.get(ATTR_DEVICE_CLASS) == DEVICE_CLASS_TEMPERATURE
+
+    entry = registry.async_get("sensor.home_realfeel_temperature")
+    assert entry
+    assert entry.unique_id == "0123456-realfeeltemperature"
+
+    state = hass.states.get("sensor.home_uv_index")
+    assert state
+    assert state.state == "6"
+    assert state.attributes.get(ATTR_ATTRIBUTION) == ATTRIBUTION
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == UV_INDEX
+    assert state.attributes.get("level") == "High"
+
+    entry = registry.async_get("sensor.home_uv_index")
+    assert entry
+    assert entry.unique_id == "0123456-uvindex"
+
+
+async def test_sensor_with_forecast(hass):
+    """Test states of the sensor with forecast."""
+    await init_integration(hass, forecast=True)
+    registry = await hass.helpers.entity_registry.async_get_registry()
+
+    state = hass.states.get("sensor.home_hours_of_sun_0d")
+    assert state
+    assert state.state == "7.2"
+    assert state.attributes.get(ATTR_ATTRIBUTION) == ATTRIBUTION
+    assert state.attributes.get(ATTR_ICON) == "mdi:weather-partly-cloudy"
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == TIME_HOURS
+
+    entry = registry.async_get("sensor.home_hours_of_sun_0d")
+    assert entry
+    assert entry.unique_id == "0123456-hoursofsun-0"
+
+    state = hass.states.get("sensor.home_realfeel_temperature_max_0d")
+    assert state
+    assert state.state == "29.8"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5985897)